### PR TITLE
feat: add openapi support to az func http project template

### LINKED ISSUE #379: Arcus.Templates.AzureFunctions.Http & OpenAPI
**Is your feature request related to a problem? Please describe.**
I always include an OpenAPI specification in my HTTP triggered Azure Functions, see: [https://www.codit.eu/blog/adding-swagger-ui-azure-functions/](https://www.codit.eu/blog/adding-swagger-ui-azure-functions/) Is it possible to add this to the Arcus.Templates.AzureFunctions.Http template?

**Describe the solution you'd like**
Maybe we can add an opt-in for this in the template. 
The easiest way to implement is probably to add the Aliencube.AzureFunctions.Extensions.OpenApi nuget package and OpenApi attributes in the template function (also some small additions to the local.settings.json and host.json file), see [https://www.codit.eu/blog/adding-swagger-ui-azure-functions/](https://www.codit.eu/blog/adding-swagger-ui-azure-functions/).

**Describe alternatives you've considered**
Not adding it 😉 

**Additional context**
none

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds OpenAPI document generation and UI to the Azure Functions HTTP trigger project template which can be opt-out.

Closes #379

### CODE REVIEW

**Comment by stijnmoreels** [comment]
Currently on 'draft' because it's using a preview package.

File: src/Arcus.Templates.AzureFunctions.Http/HttpCorrelationOpenApiResponseHeaders.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                    Required = true,
+                    Description = "Transaction ID is used to correlate multiple operation calls. A new transaction ID will be generated if request did not specified one."
+                }
+            },
+            {
+                "RequestId",
+                new OpenApiHeader
+                {
+                    Required = true,
+                    Description = "Operation ID is used to identity a single operation call."
```
Comment (fgheysels) [inline src/Arcus.Templates.AzureFunctions.Http/HttpCorrelationOpenApiResponseHeaders.cs:30]:
```suggestion
                    Description = "Operation ID is used to identify a single operation call."
```

File: src/Arcus.Templates.AzureFunctions.Http/HttpCorrelationOpenApiResponseHeaders.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        /// Gets or sets the collection of the <see cref="T:Microsoft.OpenApi.Models.OpenApiHeader" /> instances.
+        /// </summary>
+        public Dictionary<string, OpenApiHeader> Headers { get; set; } = new Dictionary<string, OpenApiHeader>
+        {
+            {
+                "X-Transaction-Id",
+                new OpenApiHeader
+                {
+                    Required = true,
+                    Description = "Transaction ID is used to correlate multiple operation calls. A new transaction ID will be generated if request did not specified one."
```
Comment (fgheysels) [inline src/Arcus.Templates.AzureFunctions.Http/HttpCorrelationOpenApiResponseHeaders.cs:22]:
```suggestion
                    Description = "Transaction ID is used to correlate multiple operation calls. A new transaction ID will be generated if no `X-Transaction-Id` header was present in the request."
```

**Review by fgheysels — COMMENTED** [review]
Just some improvements (imho) in certain descriptions.

File: src/Arcus.Templates.AzureFunctions.Http/.template.config/ide.host.json [hunk 3]
```diff
@@ -15,7 +15,14 @@
       "name": {
         "text": "Include Health Checks"
       },
-      "isVisible": true 
+      "isVisible": true
+    },
+    {
+      "id": "exclude-openapi",
+      "name": {
+        "text": "Exclude OpenApi"
```
Comment (fgheysels) [inline src/Arcus.Templates.AzureFunctions.Http/.template.config/ide.host.json:23]:
```suggestion
        "text": "Exclude OpenApi Documentation"
```
That's maybe a bit more descriptive ?

**Review by fgheysels — APPROVED** [review]

File: src/Arcus.Templates.AzureFunctions.Http/.template.config/ide.host.json (hunk 3 above)
Comment (stijnmoreels) [reply src/Arcus.Templates.AzureFunctions.Http/.template.config/ide.host.json:23]: Yeah, that would be great 😄 .

**Pull request merged**